src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -1638,6 +1628,15 @@ CHIP_ERROR GroupDataProviderImpl::GetKeySet(chip::FabricIndex fabric_index, uint
     out_keyset.epoch_keys[0].start_time = keyset.operational_keys[0].start_time;
     out_keyset.epoch_keys[1].start_time = keyset.operational_keys[1].start_time;
     out_keyset.epoch_keys[2].start_time = keyset.operational_keys[2].start_time;
+
+    // Re-derive privacy keys to minimize storage
+    for (size_t i = 0; i < keyset.keys_count; ++i)
+    {
+        ByteSpan encryption_key(keyset.operational_keys[i].encryption_key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES);
+        MutableByteSpan privacy_key(keyset.operational_keys[i].privacy_key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES);
+        ReturnErrorOnFailure(Crypto::DeriveGroupPrivacyKey(encryption_key, privacy_key));
```

Comment:
I don't understand this block.  `keyset` does not escape this function.  It's not used after this block.  Why are we modifying anything about `keyset` here?

Reply:
Removed and moved to Deserialize.